### DISCUSSION TITLE
Access logging

### DISCUSSION THREAD

**Discussion by tbonham** [dotnet/yarp#483] [Q&A] [answered]
Well issue #158 cover logging of request coming into YARP? 

I'm just wondering if I need to do all the request logging myself or if YARP will handle it?

**Comment by Tratcher** [comment, marked as answer]
What fields do you want logged? Asp.Net already logs the protocol version, url, content-type, and length. It does not log all headers or body.

**Reply by tbonham** [reply]
> What fields do you want logged? Asp.Net already logs the protocol version, url, content-type, and length. It does not log all headers or body.

I was not aware of that. I'm just coming over to kestrel, I have always used IIS. 

With my testing of kestrel, it isn't creating a log file. Is this another middle ware idea that I need to add?  

I would like to be able to have a way to log the header for debugging reasons.

**Reply by Tratcher** [reply]
See https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-3.1#configure-logging
You need to turn the "Microsoft" logs up to "Information" to get request level logs.

**Reply by Tratcher** [reply]
The framework avoids bulk logging header as they may contain sensitive information like credentials. We recommend you log only the headers you care about in your own middleware.

**Reply by tbonham** [reply]
> See https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-3.1#configure-logging
> You need to turn the "Microsoft" logs up to "Information" to get request level logs.

@Tratcher Thank you for the documentation. 

I reading the documentation. I think it going to take me sometime to get a log file the way I need it to be to log the traffic. 

```
127.0.0.1 - - [23/Jun/2020:01:38:17 +0000] "GET / HTTP/1.1" 200 396 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.106 Safari/537.36 Edg/83.0.478.54"
127.0.0.1 - - [23/Jun/2020:01:38:17 +0000] "GET /favicon.ico HTTP/1.1" 404 197 "http://127.0.0.2/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.106 Safari/537.36 Edg/83.0.478.54"
```
I thought it was going to be a little simpler than it is.

**Reply by davidfowl** [reply]
There's no log file like this built into Kestrel.

cc @shirhatti another +1 for w3c logs
